src/payment/strategies/stripev3/stripev3-payment-strategy.spec.ts

```diff
@@ -521,6 +614,106 @@ describe('StripeV3PaymentStrategy', () => {
+            const response = await strategy.execute(getStripeV3OrderRequestBodyMock());
+
+            expect(stripeV3JsMock.confirmCardPayment).toHaveBeenCalledWith(
+                'myToken',
+                {
+                    payment_method: {
+                        card: cardElement,
+                        billing_details: getStripeBillingAddress(),
+                    },
+                }
+            );
+            expect(orderActionCreator.submitOrder).toHaveBeenCalled();
+            expect(paymentMethodActionCreator.loadPaymentMethod).toHaveBeenCalled();
+            expect(paymentActionCreator.submitPayment).toHaveBeenCalled();
+            expect(response).toBe(store.getState());
+        });
+
+        it('creates the order and submit payment with a guest user without a phone number provided', async () => {
+            const elements = stripeV3JsMock.elements();
+            const cardElement = elements.create(StripeElementType.CreditCard, {});
+            const shippingAddressWithoutPhone = {
+                firstName: 'Test',
+                lastName: 'Tester',
+                company: 'Bigcommerce',
+                address1: '12345 Testing Way',
+                address2: '',
+                city: 'Some City',
+                stateOrProvince: 'California',
+                stateOrProvinceCode: 'CA',
+                country: 'United States',
+                countryCode: 'US',
+                postalCode: '95555',
+                shouldSaveAddress: true,
+                customFields: [],
+            };
+            const billingAddressWithoutPhone = {
+                id: '55c96cda6f04c',
+                firstName: 'Test',
+                lastName: 'Tester',
+                email: 'test@bigcommerce.com',
+                company: 'Bigcommerce',
+                address1: '12345 Testing Way',
+                address2: '',
+                city: 'Some City',
+                stateOrProvince: 'California',
+                stateOrProvinceCode: 'CA',
+                country: 'United States',
+                countryCode: 'US',
+                postalCode: '95555',
+                shouldSaveAddress: true,
+                customFields: [],
+            };
+
+            stripeV3JsMock.confirmCardPayment = jest.fn(
+                () => Promise.resolve(getConfirmPaymentResponse())
+            );
+
+            jest.spyOn(stripeScriptLoader, 'load').mockReturnValue(Promise.resolve(stripeV3JsMock));
+            jest.spyOn(stripeV3JsMock, 'elements').mockReturnValue(elements);
+            jest.spyOn(stripeV3JsMock.elements(), 'create').mockReturnValue(cardElement);
+            jest.spyOn(store.getState().customer, 'getCustomer').mockReturnValue(undefined);
+            jest.spyOn(store.getState().shippingAddress, 'getShippingAddress').mockReturnValue(shippingAddressWithoutPhone);
+            jest.spyOn(store.getState().billingAddress, 'getBillingAddress').mockReturnValue(billingAddressWithoutPhone);
```

Comment:
I think there's no need to mock the shipping address as you are only testing the billing address.
```suggestion
            jest.spyOn(store.getState().billingAddress, 'getBillingAddress').mockReturnValue({ ...getBillingAddress(), phone: '' });
```